### PR TITLE
Refactor cluster-mode-example with ShardingSphereDriver

### PR DESCRIPTION
**Pull request by Pace2Car**
For #22469.

Changes proposed in this pull request:
  - Refactor cluster-mode-example with ShardingSphereDriver
  - Remove Spring configuration example

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-encrypt.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      namespace: demo_spring_boot_ds_encrypt
+      server-lists: localhost:2181
+
+dataSources:
+  ds_encrypt:
+    dataSourceClassName: com.zaxxer.hikari.HikariDataSource
+    driverClassName: com.mysql.jdbc.Driver
+    jdbcUrl: jdbc:mysql://localhost:3306/demo_ds?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
+    username: root
+    password: 123456
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-encrypt.yaml:32]: Please remove password value.

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    readwrite_ds:
+      staticStrategy:
+        writeDataSourceName: write_ds
+        readDataSourceNames: 
+          - read_ds_0 
+          - read_ds_1
+      loadBalancerName: round_robin
+  loadBalancers:
+    round_robin:
+      type: ROUND_ROBIN
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml:60]: Please keep a blank line at the end.

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+mode:
+  type: Cluster
+  repository:
+    type: ZooKeeper
+    props:
+      namespace: demo_spring_boot_readwrite_ds
+      server-lists: localhost:2181
+      
+databaseName: sharding_databases_tables
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml:26]: `sharding_databases_tables` is not suitable for readwrite-splitting scenarios

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-shadow.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      namespace: demo_spring_boot_ds_shadow
+      server-lists: localhost:2181
+
+dataSources:
+  ds:
+    dataSourceClassName: com.zaxxer.hikari.HikariDataSource
+    driverClassName: com.mysql.jdbc.Driver
+    jdbcUrl: jdbc:mysql://localhost:3306/demo_ds?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
+    username: root
+    password: 123456
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-shadow.yaml:32]: Please remove password value.

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-sharding-databases-tables.yaml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      
+databaseName: sharding_databases_tables
+
+dataSources:
+  ds_0:
+    dataSourceClassName: com.zaxxer.hikari.HikariDataSource
+    driverClassName: com.mysql.jdbc.Driver
+    jdbcUrl: jdbc:mysql://localhost:3306/demo_ds_0?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
+    username: root
+    password: 123456
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-sharding-databases-tables.yaml:34]: Remove password value please.

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-namespace-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+mode:
+  type: Cluster
+  repository:
+    type: ZooKeeper
+    props:
+      namespace: demo_spring_boot_readwrite_ds
+      server-lists: localhost:2181
+      
+databaseName: sharding_databases_tables
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-namespace-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml:26]: confusing databaseName

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-encrypt.yaml (hunk 1 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-encrypt.yaml]: Removed

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml (hunk 2 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml]: Added.

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml (hunk 3 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml:26]: This config doesn't seem to be needed, I removed it, please check again

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-shadow.yaml (hunk 4 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-shadow.yaml]: Removed

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-sharding-databases-tables.yaml (hunk 5 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-boot-mybatis-example/src/main/resources/local-zookeeper-sharding-databases-tables.yaml]: Removed

File: examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-namespace-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml (hunk 6 above)
Comment (Pace2Car) [reply examples/shardingsphere-jdbc-example/single-feature-example/cluster-mode-example/cluster-mode-spring-namespace-mybatis-example/src/main/resources/local-zookeeper-readwrite-splitting.yaml]: Removed too.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**